calcom/cal.com
# fix: Routing Forms, Number operators were using string operands

## What does this PR do?
- Makes sure that number operators receives input as number instead of string.
- Implements missing >, >=, <, <= operators for Reporting.

Fixes #9170 Fixes CAL-1806

## Demo
[After the fix demo](https://www.loom.com/share/4f981c023e1b461a8aa862d995ff02f4)

## Type of change



- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?



- [x] Add a number type field say `age`.
	- Choose age greater than 50, and select router action as a custom page(You can set the custom page message as 'the age is greater than 50')
	- Go to Routing Form and input 100. Submit the form and you would see the Fallback Message instead when infact 100 > 50. This issue is fixed with the PR.
- [x] Similarly, for reporting, add a filter of age > 50 and check if you see the just added response with age 100. 
	- Now, add a filter of age < 50, you would not see the recently added response of age 50. But in the main you would see it.
- [x] Do a 'Test Preview' and make sure that adding a 100 their shows the custom page, 'the age is greater than 50'
## Checklist


- I haven't added tests that prove my fix is effective or that my feature works
